test/11_array.jl

```diff
@@ -650,4 +650,24 @@ end
     @test unique(x) == collect(1:10)
 end
 
+using JSON3
+using StructTypes
+struct MyCustomType
```

Comment:
Is a custom type really needed to test writing? Can't you just use a `CategoricalVector{String}` directly?